### PR TITLE
Exchange keys and encrypt messages.

### LINKED ISSUE #22: Exchange keys between users and encrypt messages.
The flagship RightMesh app would be remiss without encryption. We should do a key exchange when peers handshake, store if we've successfully exchanged keys in the database, check that flag when sending messages of any kind (and maybe store that in the online users list if we're going to query it a lot so we aren't making a million DB queries), and send encrypted message if we've done a successful key exchange.

### PR DESCRIPTION
**Pull request by FrazerSeymour**
With this PR, we  exchange keys with peers as they are discovered, and send encrypted messages to peers we have previously exchanged keys with.

Fixes #22

### CODE REVIEW

**Comment by ghost** [comment]
Everything looks good :+1:

File: app/src/main/java/io/left/meshim/controllers/RightMeshController.java [hunk 1]
```diff
@@ -241,6 +256,33 @@ private void handleDataReceived(RightMeshEvent e) {
...
+            User user = users.get(event.peerUuid);
+            if (user != null) {
+                user.setIsEncrypted(true);
+                users.put(event.peerUuid, user);
+                updateInterface();
+            }
+        } else if (event.status == MeshManager.FAILURE) {
+            // Key exchange has timed out, try again.
+            try {
+                meshManager.exchangeKeys(event.peerUuid, MESH_PORT, 2000);
```
Comment (ConnorHeidema) [inline app/src/main/java/io/left/meshim/controllers/RightMeshController.java:277]: Whats the purpose of port 2000. Should this be defined somewhere as TEST_PORT or something of that nature?

Comment (FrazerSeymour) [reply app/src/main/java/io/left/meshim/controllers/RightMeshController.java:277]: `2000` is a timeout time in milliseconds.

Comment (FrazerSeymour) [reply app/src/main/java/io/left/meshim/controllers/RightMeshController.java:277]: Could still store it somewhere I guess. :blush:

**Pull request merged**